File: kotlin/src/main/kotlin/com/adaptionsoft/games/trivia/runner/GameRunner.kt
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import java.util.Random


fun main(vararg args: String) {
    var notAWinner: Boolean;
    val aGame = Game()
    aGame.add("Chet")
    aGame.add("Pat")
    aGame.add("Sue")

    val rand = Random()

    do {
        aGame.roll(rand.nextInt(5) + 1)
        if (rand.nextInt(9) == 7) {
            notAWinner = aGame.wrongAnswer()
        } else {
            notAWinner = aGame.wasCorrectlyAnswered()
        }
    } while (notAWinner)
}

File: C++/src/GameRunner.cpp
#include <stdlib.h>
#include "Game.h"



void runGame() {
    Game aGame;

    aGame.add("Chet");
    aGame.add("Pat");
    aGame.add("Sue");

    bool notAWinner;

    do
    {

        aGame.roll(rand() % 5 + 1);

        if (rand() % 9 == 7)
        {
            notAWinner = aGame.wrongAnswer();
        }
        else
        {
            notAWinner = aGame.wasCorrectlyAnswered();
        }
    } while (notAWinner);
}
